### DISCUSSION TITLE
Custom Plugins, using web implementation for iOS/Android

### DISCUSSION THREAD

**Discussion by gnesher** [ionic-team/capacitor#3835] [Q&A]
Hi,

Until recently plugins had constructor property called platforms which supposedly enabled the web implementation to be used for other platforms (iOS/Android).

The property is now marked as deprecated and I was wondering what led to this decision and if there's a simple way to achieve the same results. 

For example, web share API, especially in Android/Chrome should offer feature parity with the native share (from what I can tell), if we want to switch using it in the future what's the migration path?

**Comment by aparajita** [comment]
Yes, good question. I'm about to release an enhancement to plugins that lets us easily mix and match web/native code.

**Comment by jcesarmobile** [comment]
Plugins are changing a lot on v3 and a few things are being deprecated, but you can still use the web implementation like this:

https://github.com/ionic-team/capacitor-plugins/blob/main/motion/src/index.ts#L7-L11
